core/orchestrator.go

```diff
@@ -127,36 +127,23 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 		return fmt.Errorf("orchestrator is inactive, cannot process payments")
 	}
 
-	var (
-		didPriceErr            bool
-		acceptablePrice        bool
-		unacceptableReceiveErr bool
-		didReceiveErr          bool
-	)
-
-	err = orch.acceptablePrice(ethcommon.BytesToAddress(payment.Sender), payment.GetExpectedPrice())
-	acceptablePriceErr, ok := err.(AcceptableError)
-	if err != nil {
-		glog.Error(err)
-		didPriceErr = true
+	priceInfo := payment.GetExpectedPrice()
 
-		if monitor.Enabled {
-			monitor.PaymentRecvError(sender.String(), string(manifestID), err.Error(), ok && acceptablePriceErr.Acceptable())
-		}
-	}
+	seed := new(big.Int).SetBytes(payment.TicketParams.Seed)
 
-	if err == nil || (ok && acceptablePriceErr.Acceptable()) {
-		acceptablePrice = true
+	priceInfoRat, err := common.RatPriceInfo(priceInfo)
+	if err != nil {
+		return fmt.Errorf("invalid expected price sent with payment - expectedPrice=%v err=%v", priceInfo, err)
```

Comment:
Impossible as this error will only occur when dividing by zero but the caller isn't necessarily aware of that. 

When `priceInfo` is `nil`  the return values are `(nil, nil)`

On another note , should this be `if priceInfoRat == nil || err != nil` instead?

Reply:
If `RatPriceInfo()` returns an error then `priceInfo.PricePerUnit = 0` and I'm not sure how useful logging the value of `priceInfo` is in that scenario since the error message will contain all the necessary information already. If we don't log `priceInfo` here then we don't need to manage the `priceInfo` and `priceInfoRat` variables since we can allocate `priceInfoRat` with a single call.

> On another note , should this be if priceInfoRat == nil || err != nil instead?

Yeah that would be a good defensive guard. Without it, a B that sends a payment with a nil `ExpectedPrice` would cause O to crash. I think we should separate the checks though: if `priceInfoRat == nil` we would return an error indicating a missing price and if `err != nil` we would return an error indicating why the expected price was invalid.

Reply:
Bump

Reply:
fixed in 70aa54910e33c7c697878520e51a588bf2cd95be